javaslang/src/main/java/javaslang/collection/Arrays2.java

```diff
@@ -0,0 +1,92 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _______
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  //  /\__\   JΛVΛSLΛNG
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/ \ /__\ \   Copyright 2014-2016 Javaslang, http://javaslang.io
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/\__\_/  \_//  \__/\_____/   Licensed under the Apache License, Version 2.0
+ */
+package javaslang.collection;
+
+import java.util.Random;
+
+/**
+ * Internal class, containing Java array manipulation helpers.
+ * Many arrays are represented as simple Objects, to avoid casts at the client site, and to work with all types of arrays, including primitive ones.
+ *
+ * @author Pap Lőrinc
+ * @since 3.0.0
+ */
+@SuppressWarnings({"unchecked", "SuspiciousArrayCast"})
+final class Arrays2 { // TODO reuse these in `Array` also
+    private static final Object[] EMPTY = {};
+
+    static <T> T[] emptyArray() { return (T[]) EMPTY; }
+    static boolean isNullOrEmpty(Object[] array) {
+        return (array == null) || (array.length == 0);
+    }
+
+    static <T> T getOrDefault(Object[] array, int i, T defaultValue) {
+        return (isNullOrEmpty(array) || (i < 0) || (i >= array.length))
+               ? defaultValue
+               : (T) getAt(array, i);
+    }
+
+    /** Create a single element array */
+    static Object[] asArray(Object element) {
+        final Object[] copy = arrayCopy(1, null, 0, 0, 0);
+        copy[0] = element;
+        return copy;
```

Comment:
I would remove this method completely and use

```java
new Object[] { element }
```

directly instead of

```java
asArray(element);
```

To many calls increase the cyclomatic complexity and make it hard for humans to read the code. I know that you like meaningful names. If the compiler optimizes it 'away' this method is ok for me but the method body is really hard to read.